OasisDEX/oasis-borrow
# Discover Oasis Filters UI Component

# [Discover Oasis Filters UI Component](https://app.shortcut.com/oazo-apps/story/6101/tbc-discover-oasis-filters-ui-component)

**Requires https://github.com/OasisDEX/oasis-borrow/pull/1522 to me merged first.**

Proof of concept of working filters. In theory, if everything goes right, it should only require updating endpoint addresses and response object type to make it work with actual data.

![image](https://user-images.githubusercontent.com/16230404/194706730-f934af9f-8ce7-4be2-9b4b-b90c5a6ca398.png)
  
## Changes 👷‍♀️

- Removed dedicated pages for each Discovery page and replaced that with `[slug]` approach from `next.js`,
- created filters for "High-Risk Positions" page,
- created handlers for API that is supposed to load data from data source that will be available somewhere in the future,
- fixed some minor issues with missing `key` props and general html structure in previous Discovery components.
  
## How to test 🧪

- Go to `/discovery/high-risk-positions`,
- change something in the filters,
- check if params from request to mock json file are updating accordingly.
